rest-service/manager_rest/app_context.py

```diff
@@ -19,23 +43,66 @@ def get_parser_context(sm=None):
 
 
 def update_parser_context(context):
-    raw_parser_context = _extract_parser_context(context)
-    resolver = dsl_parser_utils.create_import_resolver(
-            raw_parser_context['resolver_section'])
-    validate_definitions_version = raw_parser_context[
-        'validate_definitions_version']
-    current_app.parser_context = {
-        'resolver': resolver,
-        'validate_version': validate_definitions_version
-    }
+    current_app.parser_context = _extract_parser_context(context)
 
 
 def _extract_parser_context(context):
     context = context or {}
     cloudify_section = context.get(constants.CLOUDIFY, {})
+    resolver_section = cloudify_section.get(
+        constants.IMPORT_RESOLVER_KEY) or {}
+    resolver_section.setdefault('implementation',
+                                'manager_rest.app_context:ResolverWithPlugins')
+    resolver = dsl_parser_utils.create_import_resolver(resolver_section)
     return {
-        'resolver_section': cloudify_section.get(
-                constants.IMPORT_RESOLVER_KEY),
-        'validate_definitions_version': cloudify_section.get(
-                constants.VALIDATE_DEFINITIONS_VERSION, True)
+        'resolver': resolver,
+        'validate_version': cloudify_section.get(
+            constants.VALIDATE_DEFINITIONS_VERSION, True)
     }
+
+
+class ResolverWithPlugins(DefaultImportResolver):
+    """A resolver which translates plugin-style urls to file:// urls.
+
+    The URL: `plugin:cloudify-openstack-plugin:2.0.1` will be
+    translated to: `file:///opt/manager/resources/plugins/<id>`, where <id>
+    is the id of the plugin looked up for the current tenant.
+
+    Both the version and the filename are optional.
+    """
+    PREFIX = 'plugin:'
+
+    def fetch_import(self, import_url):
+        if self._is_plugin_url(import_url):
+            resolved = self._resolve_plugin_url(import_url)
+            if resolved:
+                import_url = resolved
+        return super(ResolverWithPlugins, self).fetch_import(import_url)
+
+    def _is_plugin_url(self, import_url):
+        return import_url.startswith(self.PREFIX)
+
+    def _resolve_plugin_url(self, import_url):
+        parts = import_url.replace(self.PREFIX, '').split(':')
```

Comment:
shouldn't it be replace(self.PREFIX, '', 1)?
otherwise, "plugin:cloudify-openstack-plugin:2.0.1" will become: "cloudify-openstack-2.0.1"..